src/macaron/__main__.py

```diff
@@ -187,12 +187,13 @@ def perform_action(action_args: argparse.Namespace) -> None:
             sys.exit(verify_policy(action_args))
 
         case "analyze":
-            # Check that the GitHub token is enabled.
-            gh_token = os.environ.get("GITHUB_TOKEN")
-            if not gh_token:
-                logger.error("GitHub access token not set.")
-                sys.exit(os.EX_USAGE)
-            global_config.gh_token = gh_token
+            # If the GitHub token has not already been read, try to read it here.
```

Comment:
I did not read the last line of your comment correctly and removed the error check entirely from this section. It will be added back in in the next commit.